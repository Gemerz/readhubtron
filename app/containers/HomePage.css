.main {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  width: 100%;
  height: 100%;
  color: #333;
}

.main > div {
  height: 100%;
  position: relative;
}

