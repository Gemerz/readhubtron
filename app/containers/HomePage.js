import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as TopicActions from '../actions/topic';
import ReadView from '../components/ReadView';
import Collection from '../components/Collection';
// style

import styles from './HomePage.css';


class HomePage extends Component {

  state = {
    refreshScrollTop: false
  }

  componentWillMount() {
    this.props.initTopic();
  }

  componentDidMount() {
    this.timeBeeFetch = setInterval(() => {
      if (this.props.topic.lastCursor > 0) {
        this.props.fetchLatestCollection(this.props.topic.lastCursor);
      }
    }, 3000);
  }
  componentWillUnmount() {
    clearInterval(this.timeBeeFetch);
  }

  render() {
    return (
      <div className={styles.main} data-tid="main">

        <Collection
          lastCursor={this.props.topic.lastCursor}
          count={this.props.topic.count}
          category={this.props.topic.category}
          collection={this.props.topic.collection}
          loading={this.props.topic.loading}
          moreLoading={this.props.topic.moreLoading}
          setCurrentUrl={this.props.setTopicCurrentUrl}
          setTopicTotalUrls={this.props.setTopicTotalUrls}
          initList={this.props.initTopic}
          loadMoreList={this.props.loadMoreTopic}
          setting={this.props.setting}
          fetchLatestCollection={this.props.fetchLatestCollection}
        />


        <ReadView
          disabledJavascript={this.props.setting.disabledJavascript}
          totalUrls={this.props.topic.totalUrls}
          setting={this.props.setting}
          currentUrl={this.props.topic.currentUrl}
        />
      </div>

    );
  }
}
function mapStateToProps(state) {
  return {
    topic: state.topic,
    setting: state.setting
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(TopicActions, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
