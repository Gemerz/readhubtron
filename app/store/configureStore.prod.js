import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, autoRehydrate } from 'redux-persist';
import { createBrowserHistory } from 'history';
import { routerMiddleware } from 'react-router-redux';
import rootReducer from '../reducers';

const history = createBrowserHistory();
const router = routerMiddleware(history);
const enhancer = applyMiddleware(thunk, router);

function configureStore(initialState) {
  const store = createStore(rootReducer, initialState, enhancer, compose(autoRehydrate()));

  persistStore(store, { blacklist: ['topic', 'router', 'news', 'tech'] }, () => {
    console.log('rehydration complete');
    store.getState();
  });
  return store;
}

export default { configureStore, history };
