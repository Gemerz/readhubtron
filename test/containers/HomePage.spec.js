import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import { ConnectedRouter } from 'react-router-redux';
import HomePage from '../../app/containers/HomePage';
import { configureStore } from '../../app/store/configureStore';

function setup(initialState) {
  const store = configureStore(initialState);
  const history = createBrowserHistory();
  const app = mount(
    <Provider store={store}>
      <ConnectedRouter history={history} >
        <HomePage />
      </ConnectedRouter>
    </Provider>
  );
  return {
    app
  };
}

describe('containers', () => {
  describe('App', () => {
    it('should display initial count', () => {
      const { app } = setup;
      console.log(app)
    });
  });
});
