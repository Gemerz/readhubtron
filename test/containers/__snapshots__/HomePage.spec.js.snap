// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers App should display initial count 1`] = `undefined`;
